lib/improver/tests/utilities/cube_manipulation/test_merge_cubes.py

```diff
@@ -223,6 +225,29 @@ def test_one_threshold_data(self):
         self.assertEqual(enuk_prob.data.shape, (1, 3, 3))
         self.assertEqual(result.data.shape, (2, 3, 3))
 
+    def test_mismatched_time_bounds_ranges(self):
+        """Test for mismatched bounds ranges error."""
+        frt = dt(2017, 11, 9, 21, 0)
+        times = [dt(2017, 11, 10, 3, 0),
+                 dt(2017, 11, 10, 4, 0),
+                 dt(2017, 11, 10, 5, 0)]
+        time_bounds = np.array([
+            [dt(2017, 11, 10, 2, 0), dt(2017, 11, 10, 3, 0)],
+            [dt(2017, 11, 10, 3, 0), dt(2017, 11, 10, 4, 0)],
+            [dt(2017, 11, 10, 2, 0), dt(2017, 11, 10, 5, 0)]])
+
+        cubelist = iris.cube.CubeList([])
+        for tpoint, tbounds in zip(times, time_bounds):
+            cube = set_up_probability_cube(
+                0.6*np.ones((2, 3, 3), dtype=np.float32),
+                np.array([278., 280.], dtype=np.float32),
+                time=tpoint, frt=frt, time_bounds=tbounds)
+            cubelist.append(cube)
+
+        msg = "Cube with mismatching time bounds ranges"
+        with self.assertRaisesRegex(ValueError, msg):
+            merge_cubes(cubelist, blend_coord="time")
+
```

Comment:
This was the first test I wrote before factoring out the new function.  I think it's a good idea to keep, as although it tests the same functionality, it would also fail if the contributors to merge_cubes were refactored in such a way that `_check_bounds_ranges` didn't change but the merged outputs did.  I know we come across this in other places, but often those groups of functions are managed as plugins with CLI tests - whereas in this case it's a collection of not-obviously-grouped utilities, so I think it's sensible to have the end-to-end as well as the component test in here.